homeassistant/components/switchbee/coordinator.py

```diff
@@ -0,0 +1,85 @@
+"""SwitchBee integration Coordinator."""
+
+from datetime import timedelta
+import logging
+
+from switchbee.api import CentralUnitAPI, SwitchBeeError
+from switchbee.device import DeviceType, SwitchBeeBaseDevice
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SwitchBeeCoordinator(DataUpdateCoordinator[dict[int, SwitchBeeBaseDevice]]):
+    """Class to manage fetching Freedompro data API."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        swb_api: CentralUnitAPI,
+        scan_interval: int,
+    ) -> None:
+        """Initialize."""
+        self._api: CentralUnitAPI = swb_api
+        self._reconnect_counts: int = 0
+        self._mac_addr_fmt: str = format_mac(swb_api.mac)
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=scan_interval),
+        )
+
+    @property
+    def api(self) -> CentralUnitAPI:
```

Comment:
Don't use properties that just return an instance attribute. An instance attribute is enough.